File: AI-Career-Coach-frontend/src/modules/mock-interview-v2/types/evaluation.ts
export interface MetricScoreV2 {
  name: string;
  score: number;
  description: string;
  benchmark: string;
}

export interface TurnEvaluationV2 {
  question_number: number;
  phase: string;
  question_text: string;
  candidate_answer: string;
  technical_accuracy: number;
  communication_clarity: number;
  completeness: number;
  strengths: string[];
  weaknesses: string[];
  suggested_improvement: string;
  overall_turn_score: number;
}

export interface FinalEvaluationReportV2 {
  session_id: string;
  candidate_name: string;
  target_role: string;
  interviewer_name: string;
  interview_type: string;
  difficulty: string;
  total_duration_seconds: number;
  completed_at: string;
  
  overall_score: number;
  technical_score: number;
  communication_score: number;
  problem_solving_score: number;
  confidence_score: number;
  professionalism_score: number;
  relevance_score: number;
  clarity_score: number;
  
  metrics_breakdown: MetricScoreV2[];
  hiring_recommendation: "Strong Hire" | "Hire" | "Leaning Hire" | "Do Not Hire" | string;
  recommendation_summary: string;
  
  top_strengths: string[];
  areas_for_improvement: string[];
  technical_gaps: string[];
  turn_evaluations: TurnEvaluationV2[];
}

File: AI-Career-Coach-Backend/app/mock_interview_v2/schemas/evaluation.py
from typing import Dict, Any, List, Optional
from pydantic import BaseModel, Field


class TurnEvaluationMetric(BaseModel):
    score: int = Field(..., ge=0, le=100)
    feedback: str


class TurnEvaluation(BaseModel):
    question_number: int
    phase: str
    question_text: str
    candidate_answer: str
    technical_accuracy: int
    communication_clarity: int
    completeness: int
    strengths: List[str] = []
    weaknesses: List[str] = []
    suggested_improvement: str
    overall_turn_score: int


class MetricScore(BaseModel):
    name: str
    score: int
    description: str
    benchmark: str = "Industry Standard (75%)"


class FinalEvaluationReportV2(BaseModel):
    session_id: str
    candidate_name: str
    target_role: str
    interviewer_name: str
    interview_type: str
    difficulty: str
    total_duration_seconds: int
    completed_at: str
    
    # 8 Core Recruitment Metrics
    overall_score: int
    technical_score: int
    communication_score: int
    problem_solving_score: int
    confidence_score: int
    professionalism_score: int
    relevance_score: int
    clarity_score: int
    
    # Granular Breakdown
    metrics_breakdown: List[MetricScore]
    hiring_recommendation: str  # Strong Hire, Hire, Leaning Hire, Do Not Hire
    recommendation_summary: str
    
    top_strengths: List[str]
    areas_for_improvement: List[str]
    technical_gaps: List[str]
    strongest_answer_excerpt: Optional[str] = None
    weakest_answer_excerpt: Optional[str] = None
    
    turn_evaluations: List[TurnEvaluation] = []
